### DISCUSSION TITLE
Custom icons with entity selection

### DISCUSSION THREAD

**Discussion by roycw** [geoman-io/leaflet-geoman#1503] [Questions & Help]
HI Guys,
Need soem guidance on how ic an imeplemnmt this. 
i need to be able to add 3 'types' of markers' 
but before i add each marker I need to select the entity id.
e.g marker 1: peros, marker 2 building
when i click on the person marker i got a popover that will let me sleect from a list of items. then i can place that cusotm marker on the map.

I know how to do all the wiring but not asure how can i implement it in the ui level. can i use the current toolbar to open a popup etc...

Any guidance on how to start

**Comment by Falke-Design** [comment]
You can add a custom button with a custom action to the toolbar:
```
map.pm.Toolbar.createCustomControl({
  name: 'alertBox',
  block: 'custom',
  className: 'leaflet-pm-icon-marker xyz-class',
  title: 'Count layers',
  onClick: () => {
    alert(`There are ${L.PM.Utils.findLayers(map).length} layers on the map`);
  },
  toggle: false,
});
```
